File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    // 1. 각 작업 완료까지 걸리는 날짜 계산
    let days = progresses.map((p, i) => {
        return Math.ceil((100 - p) / speeds[i]);
    });
    
    // 2. 배포 묶기
    let currentDay = days[0];
    let count = 1;
    
    for (let i = 1; i < days.length; i++) {
        if (days[i] <= currentDay) {
            count++;
        } else {
            answer.push(count);
            currentDay = days[i];
            count = 1;
        }
    }
    
    // 마지막 배포
    answer.push(count);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.kt
class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        val result = mutableListOf<Int>()

        // 1. 각 작업 완료까지 필요한 날짜 계산
        val days = IntArray(progresses.size)
        for (i in progresses.indices) {
            days[i] = (100 - progresses[i] + speeds[i] - 1) / speeds[i]
        }

        // 2. 배포 묶기
        var currentDay = days[0]
        var count = 1

        for (i in 1 until days.size) {
            if (days[i] <= currentDay) {
                count++
            } else {
                result.add(count)
                currentDay = days[i]
                count = 1
            }
        }

        // 마지막 배포
        result.add(count)

        return result.toIntArray()
    }
}
